### ISSUE TITLE
stack haddock's top page has bad links ?

### ISSUE THREAD

**Issue by simonmichael** [commercialhaskell/stack#419]
I ran `stack haddock --no-haddock-deps` in the hledger repo:
```
Generating Haddock index in
file:///Users/simon/src/hledger/.stack-work/install/x86_64-osx/nightly-2015-06-17/7.10.1/doc/index.html
```
The links in that page are broken, eg the Hledger module links to `Hledger.html` but the file is actually at `hledger-lib-0.25.98/Hledger.html`.

**Comment by snoyberg** [comment]
@manny-fp Can you take a look?

**Comment by manny-fp** [comment]
There seems to be a difference between the way GHC 7.10's haddock haves from GHC 7.8's.  The `haddock --gen-contents` appears to be ignoring the relative paths I specify in the `--read-interface` arguments (but is reading the interface files).  Oddly, `haddock --gen-index` does seem to be using them properly.

**Comment by manny-fp** [comment]
Another data point: `cabal --enable-documentation`'s index seems to have exactly the same problem.

**Comment by manny-fp** [comment]
Ok, this looks like a [bug](https://ghc.haskell.org/trac/ghc/ticket/10206) in the version of `haddock` that ships with GHC 7.10.1.  I reproduced using just simple `haddock` commands on OS X and Linux.  The bug report indicates it's fixed for GHC 7.10.2, and I confirmed that the problem is solved with the current HEAD of the haddock repo.

I think we should close this issue and wait for GHC 7.10.2.  The alternative would be having stack use a custom patched haddock binary, which seems excessive (the fixed version of haddock is not even on hackage, so no quick and easy way to install it).
